From c4265cb013238c402026abb4b0f1248c72d3420c Mon Sep 17 00:00:00 2001
From: Prasanna Kerekoppa <prasanna.kerekoppa@cypress.com>
Date: Wed, 4 Nov 2020 22:37:13 -0600
Subject: [PATCH 135/138] brcmfmac: To fix crash when platform does not contain
 platform data/DTS

dev_get_platdata() will return NULL in this case which causes the NULL access
Some snippet of the dump below
[  298.890591] Hardware name: Freescale i.MX8QXP MEK (DT)	22
[  298.895734] task: ffff80083662a880 task.stack: ffff000028358000	23
[  298.901865] PC is at brcmf_common_pd_probe+0x2c/0x68 [brcmfmac]	24
[  298.907963] LR is at brcmf_common_pd_probe+0x24/0x68 [brcmfmac]	25
[  298.913883] pc : [<ffff000000eee02c>] lr : [<ffff000000eee024>] pstate: 40000145	26
[  298.921283] sp : ffff00002835baa0

Signed-off-by: Prasanna Kerekoppa <prasanna.kerekoppa@cypress.com>
Signed-off-by: Chi-hsien Lin <chi-hsien.lin@infineon.com>
---
 drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c
index f2f6d46..2825d65 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c
@@ -534,7 +534,7 @@ static int __init brcmf_common_pd_probe(struct platform_device *pdev)
 
 	brcmfmac_pdata = dev_get_platdata(&pdev->dev);
 
-	if (brcmfmac_pdata->power_on)
+	if (brcmfmac_pdata && brcmfmac_pdata->power_on)
 		brcmfmac_pdata->power_on();
 
 	return 0;
-- 
1.9.1

